### PR TITLE
Add PEP420 namespace support.

### PR DESCRIPTION
**Pull request by CnlPepper**
Should address issue #2918.

### CODE REVIEW

File: Cython/Utils.py [hunk 1]
```diff
@@ -148,21 +148,31 @@ def find_root_package_dir(file_path):
...
 @cached_function
-def is_package_dir(dir_path):
+def contains_init(dir_path):
     for filename in PACKAGE_FILES:
         path = os.path.join(dir_path, filename)
         if path_exists(path):
             return 1
 
 
+@cached_function
```
Comment (scoder) [inline Cython/Utils.py]: I don't think we need this decorator here. It's enough to have it on `contains_init()`.

File: Cython/Utils.py [hunk 2]
```diff
@@ -148,21 +148,31 @@ def find_root_package_dir(file_path):
 
 @cached_function
 def check_package_dir(dir, package_names):
```
Comment (scoder) [inline Cython/Utils.py]: `dir` is not an ideal name since it shadows the builtin. Better change it along the way. The function below uses `dir_path`.

**Comment by scoder** [comment]
The test can be disabled for Py2 by adding it to the `VER_DEP_MODULES` exclusions in `runtests.py`, section `(3,3)`.

File: Cython/Utils.py (hunk 2 above)
Comment (CnlPepper) [reply Cython/Utils.py]: Renamed dir -> dir_path.

File: Cython/Utils.py (hunk 1 above)
Comment (CnlPepper) [reply Cython/Utils.py]: Removed decorator.

**Comment by scoder** [comment]
Thanks

**Pull request merged**